pkg/config/validation/validation.go

```diff
@@ -2896,3 +2896,46 @@ func validateLocalities(localities []string) error {
 
 	return nil
 }
+
+// ValidateMeshNetworks validates meshnetworks.
+func ValidateMeshNetworks(meshnetworks *meshconfig.MeshNetworks) (errs error) {
+	for name, network := range meshnetworks.Networks {
+		if err := validateNetwork(network); err != nil {
+			errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("invalid network %v:", name)))
+		}
+	}
+	return
+}
+
+func validateNetwork(network *meshconfig.Network) (errs error) {
+	for _, n := range network.Endpoints {
+		switch e := n.Ne.(type) {
+		case *meshconfig.Network_NetworkEndpoints_FromCidr:
+			if err := ValidateIPSubnet(e.FromCidr); err != nil {
+				errs = multierror.Append(errs, err)
+			}
+		case *meshconfig.Network_NetworkEndpoints_FromRegistry:
+			r := e.FromRegistry
+			if strings.ToLower(r) == "kubernetes" && r != "Kubernetes" {
+				errs = multierror.Append(errs, fmt.Errorf("registry name should be Kubernetes instead of %v", r))
```

Comment:
this is technically breaking possible behavior. registry is an arbitrary string based, it could be `kubernetes` to refer to that registry rather than Kubernetes.

Reply:
so are you saying this is not a validation we can do by just looking at the resource itself? we need to look at kubernetes secret for the remote cluster to decide?

Reply:
Yeah that is my understanding at least. They put the multicluster secrets in the cluster, then reference them here. They can have any name as far as I know

Reply:
@hzxuzhonghu @nmittler any suggestions what we can do here for meshnetworks validation?

Reply:
Yeah, not sure of a good way to do this apart from actually inspecting the cluster. Seems awfully heavy for validation ... is there any precedent for doing something like this?

Reply:
No, validation should be stateless. If we really want to do this, it belongs in istioctl analyze I think

Reply:
agreed. Does the other fields validation seem necessary here?